reana-ui/src/actions.js

```diff
@@ -268,7 +269,11 @@ export function fetchWorkflow(id) {
     if (workflow) {
       return workflow;
     } else {
-      dispatch(fetchWorkflows());
+      dispatch(
+        fetchWorkflows(...Array(5), {
```

Comment:
Phew! It took me a while to understand the `Array(5)` part. I think that for these cases we can make use of destructuring objects as function args. We can make `fetchWorkflows` to expect an object instead of multiple args.

After playing a little bit with it, I came up with this solution, let me know what you think:

```diff
diff --git a/reana-ui/src/actions.js b/reana-ui/src/actions.js
index 3d4ac03..aaa950b 100644
--- a/reana-ui/src/actions.js
+++ b/reana-ui/src/actions.js
@@ -232,14 +232,14 @@ export function confirmUserEmail(token) {
   };
 }
 
-export function fetchWorkflows(
+export function fetchWorkflows({
   pagination,
   search,
   status,
   sort,
   showLoader = true,
-  workflowId = {}
-) {
+  workflowId,
+}) {
   return async (dispatch) => {
     if (showLoader) {
       dispatch({ type: WORKFLOWS_FETCH });
@@ -270,8 +270,10 @@ export function fetchWorkflow(id) {
       return workflow;
     } else {
       dispatch(
-        fetchWorkflows(...Array(5), {
-          workflow_id_or_name: id,
+        fetchWorkflows({
+          workflowId: {
+            workflow_id_or_name: id,
+          },
         })
       );
     }
diff --git a/reana-ui/src/pages/workflowList/WorkflowList.js b/reana-ui/src/pages/workflowList/WorkflowList.js
index 98d057b..7e4b08a 100644
--- a/reana-ui/src/pages/workflowList/WorkflowList.js
+++ b/reana-ui/src/pages/workflowList/WorkflowList.js
@@ -71,20 +71,25 @@ function Workflows() {
 
   useEffect(() => {
     dispatch(
-      fetchWorkflows({ ...pagination }, searchFilter, statusFilter, sortDir)
+      fetchWorkflows({
+        pagination: { ...pagination },
+        search: searchFilter,
+        status: statusFilter,
+        sort: sortDir,
+      })
     );
 
     if (!interval.current && reanaToken && pollingSecs) {
       interval.current = setInterval(() => {
         const showLoader = false;
         dispatch(
-          fetchWorkflows(
-            { ...pagination },
-            searchFilter,
-            statusFilter,
-            sortDir,
-            showLoader
-          )
+          fetchWorkflows({
+            pagination: { ...pagination },
+            search: searchFilter,
+            status: statusFilter,
+            sort: sortDir,
+            showLoader,
+          })
         );
         setRefreshedAt(currentUTCTime());
       }, pollingSecs * 1000);

```

Perhaps it's too much of a change, but it looks cleaner to me. I haven't tested all the affected parts.

Reply:
Thanks for a nice suggestion! Indeed looks much cleaner and works as expected